### ISSUE TITLE
Follow logs

### ISSUE THREAD

**Issue by simplelife2010** [nbedos/cistern#22]
Could we please get logs updated as we view them via 'v'? Like a tail -f.

**Comment by nbedos** [comment]
I tried to implement log streaming in a few ways in earlier versions of `cistern` because that's a feature I really wanted but I couldn't find a way to do it that isn't insanely wasteful.

Travis, GitLab, CircleCI and AppVeyor all have a REST endpoint that allows downloading logs while a job is running.

To implement log streaming, the naive thing to do would be to download the whole job log every few seconds but that incurs a prohibitively high cost on the server side. For example for a job that lasts 15 minutes, assuming the log size grows linearly from 0 to 1MB and cistern downloads it every 15 seconds the total downloaded size would be about 30MB. 30 times the size of the final log file. This is very wasteful and I don't think that's acceptable, not to mention that it could be abused by setting the refresh rate even lower than once every 15 seconds.

A smarter method would be to download only the portion of the log file that has not been downloaded yet using the HTTP range header but I couldn't get this to work: it was not supported by the servers of the CI providers I tried this on (gitlab.com and Travis if I remember correctly).

Finally Azure allows access to the logs of a task only once the task is finished so for this provider we cannot even rely on the REST API for any kind of log streaming.

Personaly I just rely on `cistern` to see when a job fails and I find that I don't miss log streaming at all.
